tests/ibc/ibc_middleware_test.go

```diff
@@ -0,0 +1,814 @@
+	chainBAccount := suite.chainB.SenderAccount.GetAddress()
+
+	sendAmt := math.NewIntFromBigInt(nativeErc20.InitialBal)
+	senderEthAddr := nativeErc20.Account
+	sender := sdk.AccAddress(senderEthAddr.Bytes())
+
+	msg := transfertypes.NewMsgTransfer(
+		path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID,
+		sdk.NewCoin(nativeErc20.Denom, sendAmt),
+		sender.String(), chainBAccount.String(),
+		timeoutHeight, 0, "",
+	)
+	_, err := suite.evmChainA.SendMsgs(msg)
+	suite.Require().NoError(err) // message committed
+
+	balAfterTransfer := evmApp.Erc20Keeper.BalanceOf(evmCtx, nativeErc20.ContractAbi, nativeErc20.ContractAddr, senderEthAddr)
+	suite.Require().Equal(
+		new(big.Int).Sub(nativeErc20.InitialBal, sendAmt.BigInt()).String(),
+		balAfterTransfer.String(),
+	)
+
+	// Check native erc20 token is escrowed on evmChainA for sending to chainB.
+	escrowAddr := transfertypes.GetEscrowAddress(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+	escrowedBal := evmApp.BankKeeper.GetBalance(evmCtx, escrowAddr, nativeErc20.Denom)
+	suite.Require().Equal(sendAmt.String(), escrowedBal.Amount.String())
+
+	// chainBNativeErc20Denom is the native erc20 token denom on chainB from evmChainA through IBC.
+	chainBNativeErc20Denom := transfertypes.NewDenom(
+		nativeErc20.Denom,
+		transfertypes.NewHop(
+			suite.pathAToB.EndpointB.ChannelConfig.PortID,
+			suite.pathAToB.EndpointB.ChannelID,
+		),
+	)
+	receiver := sender // the receiver is the sender on evmChainA
+	// Mock the transfer of received native erc20 token by evmChainA to evmChainA.
+	// Note that ChainB didn't receive the native erc20 token. We just assume that.
+	packetData := transfertypes.NewFungibleTokenPacketData(
+		chainBNativeErc20Denom.Path(),
+		sendAmt.String(),
+		chainBAccount.String(),
+		receiver.String(),
+		"",
+	)
+	packet := channeltypes.Packet{
+		Sequence:           1,
+		SourcePort:         path.EndpointB.ChannelConfig.PortID,
+		SourceChannel:      path.EndpointB.ChannelID,
+		DestinationPort:    path.EndpointA.ChannelConfig.PortID,
+		DestinationChannel: path.EndpointA.ChannelID,
+		Data:               packetData.GetBytes(),
+		TimeoutHeight:      suite.evmChainA.GetTimeoutHeight(),
+		TimeoutTimestamp:   0,
+	}
+
+	transferStack, ok := suite.evmChainA.App.GetIBCKeeper().PortKeeper.Route(transfertypes.ModuleName)
+	suite.Require().True(ok)
+
+	sourceChan := path.EndpointB.GetChannel()
+	ack := transferStack.OnRecvPacket(
+		evmCtx,
+		sourceChan.Version,
+		packet,
+		suite.evmChainA.SenderAccount.GetAddress(),
+	)
+	suite.Require().True(ack.Success())
+
+	// Check un-escrowed balance on evmChainA after receiving the packet.
+	escrowedBal = evmApp.BankKeeper.GetBalance(evmCtx, escrowAddr, nativeErc20.Denom)
+	suite.Require().True(escrowedBal.IsZero(), "escrowed balance should be un-escrowed after receiving the packet")
+	balAfterUnescrow := evmApp.Erc20Keeper.BalanceOf(evmCtx, nativeErc20.ContractAbi, nativeErc20.ContractAddr, senderEthAddr)
```

Comment:
check that the bank balance is 0 for the token - should be all converted back to native token

Reply:
I just added on here,  Thanks

https://github.com/cosmos/evm/blob/7e1f25917fe16e985a3cda51622474feee3d44dc/tests/ibc/ibc_middleware_test.go#L299-L305